criteria.c

```diff
@@ -0,0 +1,65 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <wayland-client.h>
+#include "mako.h"
+#include "criteria.h"
+#include "notification.h"
+
+struct mako_criteria *create_criteria(struct mako_state *state) {
+	struct mako_criteria *criteria = calloc(1, sizeof(struct mako_criteria));
+	if (criteria == NULL) {
+		fprintf(stderr, "allocation failed\n");
+		return NULL;
+	}
+
+	// Copy the global configuration, which will then be overridden as needed
+	// during the configuration of this criteria.
+	*criteria->config = state->config;
+	return criteria;
+}
+
+void finish_criteria(struct mako_criteria *criteria) {
```

Comment:
If you have a `create_criteria` function, you need `destroy_criteria` instead of `finish_criteria`. The pattern is:

* `create` and `destroy` allocates and de-allocates the structs and their fields
* `init` and `finish` only allocates and de-allocates the fields inside the structs, not the structs themselves